Makefile

```diff
@@ -112,122 +217,103 @@ docker-build: generate-deepcopy build-for-docker .dockerflag.mk ## Build docker
-	git clone --branch v1.0.0 https://github.com/kubernetes-sigs/cluster-api.git
+	git clone --branch v1.0.0 --depth 1 https://github.com/kubernetes-sigs/cluster-api.git
 
 cluster-api/tilt-settings.json: hack/tilt-settings.json cluster-api
 	cp ./hack/tilt-settings.json cluster-api
 
-##@ End-to-End Testing
+##@ Tests
+## --------------------------------------
+## Tests
+## --------------------------------------
+
+export KUBEBUILDER_ASSETS=$(REPO_ROOT)/$(TOOLS_BIN_DIR)
+.PHONY: test
+test: generate-mocks lint $(GINKGO_V2) $(KUBECTL) $(API_SERVER) $(ETCD) ## Run tests. At the moment this is only unit tests.
+	@# The following is a slightly funky way to make sure the ginkgo statements are removed regardless the test results.
+	@$(GINKGO_V2) --label-filter="!integ" --cover -coverprofile cover.out --covermode=atomic -v ./api/... ./controllers/... ./pkg/...; EXIT_STATUS=$$?;\
+		./hack/testing_ginkgo_recover_statements.sh --remove; exit $$EXIT_STATUS
 
 CLUSTER_TEMPLATES_INPUT_FILES=$(shell find test/e2e/data/infrastructure-cloudstack/*/cluster-template*/* test/e2e/data/infrastructure-cloudstack/*/bases/* -type f)
 CLUSTER_TEMPLATES_OUTPUT_FILES=$(shell find test/e2e/data/infrastructure-cloudstack -type d -name "cluster-template*" -exec echo {}.yaml \;)
 .PHONY: e2e-cluster-templates
 e2e-cluster-templates: $(CLUSTER_TEMPLATES_OUTPUT_FILES) ## Generate cluster template files for e2e testing.
-cluster-template%yaml: bin/kustomize $(CLUSTER_TEMPLATES_INPUT_FILES)
-	kustomize build --load-restrictor LoadRestrictionsNone $(basename $@) > $@
+cluster-template%yaml: $(KUSTOMIZE) $(CLUSTER_TEMPLATES_INPUT_FILES)
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(basename $@) > $@
 
-e2e-essentials: bin/ginkgo_v1 e2e-cluster-templates kind-cluster ## Fulfill essential tasks for e2e testing.
-	IMG=$(IMG_LOCAL) make manifests docker-build docker-push
+e2e-essentials: $(GINKGO_V1) e2e-cluster-templates kind-cluster ## Fulfill essential tasks for e2e testing.
+	IMG=$(IMG_LOCAL) make generate-manifests docker-build docker-push
 
 JOB ?= .*
 run-e2e: e2e-essentials ## Run e2e testing. JOB is an optional REGEXP to select certainn test cases to run. e.g. JOB=PR-Blocking, JOB=Conformance
 	cd test/e2e && \
-	ginkgo_v1 -v -trace -tags=e2e -focus=$(JOB) -skip=Conformance -nodes=1 -noColor=false ./... -- \
-	    -e2e.artifacts-folder=${PROJECT_DIR}/_artifacts \
-	    -e2e.config=${PROJECT_DIR}/test/e2e/config/cloudstack.yaml \
+	$(REPO_ROOT)/$(GINKGO_V1) -v -trace -tags=e2e -focus=$(JOB) -skip=Conformance -nodes=1 -noColor=false ./... -- \
+	    -e2e.artifacts-folder=${REPO_ROOT}/_artifacts \
+	    -e2e.config=${REPO_ROOT}/test/e2e/config/cloudstack.yaml \
 	    -e2e.skip-resource-cleanup=false -e2e.use-existing-cluster=true
 	kind delete clusters capi-test
+
+##@ Cleanup
+## --------------------------------------
+## Cleanup
+## --------------------------------------
+
+.PHONY: clean
+clean: ## Cleans up everything.
+	rm -rf $(RELEASE_DIR)
```

Comment:
Main has the removal of generated test templates. Will probably come in with a merge.